addons/test/misra/misra-test.c

```diff
@@ -726,6 +726,12 @@ static void misra_10_4(u32 x, s32 y) {
   }
 }
 
+//10652
+bool fp(const char c) //8.4
```

Comment:
I would prefer a test in the function `misra_10_4` above. A single line should be enough. it will be easier to see what fp the test is about then. I don't think you should add a "return" statement. I have the feeling you could reproduce the false positive with `dummy = (char)'1'`